### ISSUE TITLE
Specter on Windows doesn't stop all specterd.exe instances after quitting

### ISSUE THREAD

**Issue by hodlnym** [cryptoadvance/specter-desktop#772]
Whenever Specter Desktop is started on Windows 10, for some reason it (at least in my case) runs as 4 Specter.exe processes and 2 specterd.exe processes. After the application is closed (using the Quit button in taskbar) all 4 Specter.exe processes are correctly stopped, together with one of the specterd.exe processes. The other specterd.exe however keeps running (until e.g. killed manually).

If the Specter Desktop is started again, the same thing happens, so after closing it there are now 2 specterd.exe instances hanging.

This can be very annoying especially when installing new version of Specter - after installation Specter downloads new version of specterd.exe (`"Fetching specter binaries..."`) and then fails when trying to unzip it because it cannot replace the existing file that is being blocked by the hanging process. Specter Desktop then displays error `"Error: Script failed to execute, this normally means an error was thrown. Check the renderer console for the error."` which isn't very helpful and doesn't in any way indicate that user should just kill all hanging specterd.exe instances and everything would be fine.

I noticed this behavior at first when upgrading from Specter v0.10.1 to v0.10.4, but the latest v0.10.4 behaves the same way.

**Comment by mmikeww** [comment]
i already reported this same problem in #923 and #1447 but i will confirm yet again that this problem still exists when i tried to update just down to version 2.0.2

**Comment by al-munazzim** [comment]
## Triage Assessment

**Type**: Bug (Windows process management)
**Priority**: LOW
**Verdict**: VALID-BUG (stale)

Filed Dec 2020: on Windows 10, quitting Specter leaves orphan `specterd.exe` processes. Each restart accumulates more orphans, eventually blocking updates (file locked by running process).

**Analysis:**
- Root cause: the Electron app spawns specterd as a child process but the cleanup/shutdown handler misses one instance. On Windows, child processes are not automatically killed when the parent exits (unlike Unix process groups).
- This is a genuine bug in Windows process lifecycle management.
- 1 comment from the reporter with additional details, no fix attempted.
- The Electron + specterd architecture has been stable since v2.x — this may still reproduce.

**Potential fix:**
- Use a Windows Job Object to group all child processes, ensuring they are killed when the parent exits
- Or explicitly track all spawned PIDs and `taskkill` them on shutdown
- In `electron/main.js` shutdown handler, ensure all specterd processes are terminated

**Recommendation:** Valid bug but 5+ years stale with no recent reports. Low priority — may have been inadvertently fixed by Electron/specterd changes. Worth confirming reproducibility on current version before investing effort.

---
*Automated triage by @al-munazzim — maintainer bot for specter-desktop*
